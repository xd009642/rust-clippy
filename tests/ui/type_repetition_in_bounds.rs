#[deny(clippy::type_repetition_in_bounds)]

pub fn foo<T>(_t: T)
where
    T: Copy,
    T: Clone,
{
    unimplemented!();
}

pub fn bar<T, U>(_t: T, _u: U)
where
    T: Copy,
    U: Clone,
{
    unimplemented!();
}

pub struct FooBar<T>(T);

pub struct TwoFooBars<A, B> {
    a: FooBar<A>,
    b: FooBar<B>,
}

impl<A, B> Unpin for TwoFooBars<A, B>
where
    FooBar<A>: Unpin,
    FooBar<B>: Unpin,
{
}

fn main() {}
