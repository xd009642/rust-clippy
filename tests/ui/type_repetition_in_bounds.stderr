error: this type has already been used as a bound predicate
  --> $DIR/type_repetition_in_bounds.rs:6:5
   |
LL |     T: Clone,
   |     ^^^^^^^^
   |
note: lint level defined here
  --> $DIR/type_repetition_in_bounds.rs:1:8
   |
LL | #[deny(clippy::type_repetition_in_bounds)]
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: consider combining the bounds: `T: Copy + Clone`

error: this type has already been used as a bound predicate
  --> $DIR/type_repetition_in_bounds.rs:29:5
   |
LL |     FooBar<B>: Unpin,
   |     ^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::type-repetition-in-bounds` implied by `-D warnings`
   = help: consider combining the bounds: `FooBar<B>: Unpin + Unpin`

error: aborting due to 2 previous errors

